### ISSUE TITLE
ARIA Label overwrites accessible name for table header

### ISSUE THREAD

**Issue by yatil** [Sage/carbon#3222]
### Current behaviour

If you navigate to a table using a screen reader, the screen reader will announce that the column is sortable, but not the content of the column.

### Expected behaviour

The header content is also conveyed.

### Reproducible example

https://carbon.sage.com/?path=/story/table--classic

### Suggested solution(s)

The reason is that #1248 uses `aria-label` to add instructions to the link, but `aria-label` defines the “accessible name” of an element, effectively overwriting the content in the header.

The header value, in above example “Country” or “Code” should be mentioned in the `aria-label`, too. Alternatively you can use `aria-describedby` for the description of the functionality.

**Comment by jamime** [comment]
Thanks @yatil we're in the process of removing the `Table` component and replacing it with the `FlatTable` component. We'll take a look at `FlatTable` to ensure that the same defect is not present.

 FE-3159

**Comment by nicktitchmarsh** [comment]
FE-3291 has been raised to address this

**Comment by nicktitchmarsh** [comment]
#3361 is a duplicate of this issue

**Comment by ianforster-sage** [comment]
Fixed https://github.com/Sage/carbon/pull/3450
